### PR TITLE
[Ripple] Add tests demonstrating unexpected layer.mask behavior.

### PR DESCRIPTION
**Pull request by jverkoey**
These tests demonstrate how a ripple view added as a subview can end up with an incorrect layer mask if the ripple view's frame does not match the bounds of the parent view and the parent view's layer has a shadowPath. All of the tests intentionally pass, with the final test demonstrating the incorrect behavior.

In a follow-up change, I will be adding a behavioral flag to turn off this unexpected behavior in favor of a more explicit approach to configuring the ripple view's layer mask.

Part of https://github.com/material-components/material-components-ios/issues/6913

### CODE REVIEW

File: components/Ripple/tests/unit/MDCRippleViewUsesSuperviewShadowLayerAsMaskTests.m [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import <XCTest/XCTest.h>
+
+#import "MaterialRipple.h"
+
+@interface MDCRippleViewUsesSuperviewShadowLayerAsMaskTests : XCTestCase
```
Comment (romoore) [inline components/Ripple/tests/unit/MDCRippleViewUsesSuperviewShadowLayerAsMaskTests.m:25]: Nit: Please add documentation to the interface.

File: components/Ripple/tests/unit/MDCRippleViewUsesSuperviewShadowLayerAsMaskTests.m [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+  // When
+  parentView.layer.shadowPath = [UIBezierPath bezierPathWithRect:parentView.bounds].CGPath;
+  rippleView.rippleStyle = MDCRippleStyleUnbounded;
+
+  // Then
+  XCTAssertNil(rippleView.layer.mask);
+}
+
+- (void)testInheritsParentShadowPathWhenRippleStyleSetToBoundedEvenWhenFrameDoesNotEqualBounds {
```
Comment (romoore) [inline components/Ripple/tests/unit/MDCRippleViewUsesSuperviewShadowLayerAsMaskTests.m]: Tip: Name this something like `testIncorrectlyInherits...` and add a `TODO` to track the fix.

**Review by romoore — APPROVED** [review]
LGTM with nits.

File: components/Ripple/tests/unit/MDCRippleViewUsesSuperviewShadowLayerAsMaskTests.m (hunk 1 above)
Comment (jverkoey) [reply components/Ripple/tests/unit/MDCRippleViewUsesSuperviewShadowLayerAsMaskTests.m:25]: Done.

File: components/Ripple/tests/unit/MDCRippleViewUsesSuperviewShadowLayerAsMaskTests.m (hunk 2 above)
Comment (jverkoey) [reply components/Ripple/tests/unit/MDCRippleViewUsesSuperviewShadowLayerAsMaskTests.m]: Done.

**Pull request merged**